sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_parse_id.py

```diff
@@ -0,0 +1,41 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+from ._shared import parse_key_vault_identifier, ParsedId
+
+
+def parse_certificate_id(original_id):
+    # type: (str) -> ParsedId
+    """Parses a full certificate's ID into a class.
+
+    :param str original_id: the full original identifier of a certificate
+    :returns: Returns a parsed certificate id
+    :rtype: ~azure.keyvault.certificates.ParsedId
+    :raises: ValueError
+
+    Example:
+        .. literalinclude:: ../tests/test_parse_id.py
+            :start-after: [START parse_certificate_id]
+            :end-before: [END parse_certificate_id]
+            :language: python
+            :caption: Parse a certificate's ID
+            :dedent: 8
+    """
+    parsed_id = parse_key_vault_identifier(original_id)
+
+    valid_collections = ["certificates", "deletedcertificates"]
+
+    if parsed_id.collection not in valid_collections:
```

Comment:
I don't see a value in asserting the collection. I had the same feedback for JS. IS there some reason we care if they pass any URL since we're just parsing? This feels like client-side validation, which we're not supposed to do unless it's critical for client operations like forming a correct canonical URL (REST operation path,  for e.g.). Even then, only basic null or empty checks. Let the server validate.

Reply:
Thanks so much @heaths! Since removing collection asserting would make the method the same for keys, secrets, and certificates, are you ok if I just expose it as `parse_id`?

Reply:
For Java and .NET that won't work since it'll cause strongly-typed name collisions across disparate libraries, so we need to have qualifiers. As such, the other languages should to. Besides, this gives us the opportunity later to assert collections if it causes customer confusion.